rfc/rfc-0005-customizable-consumption-offset.md

```diff
@@ -0,0 +1,68 @@
+# RFC-0005: Optional starting offset for processor inputs
+
+**Authors:** Eric Bottard
+
+**Status:**
+
+**Pull Request URL:** https://github.com/projectriff/riff/pull/1367
+
+**Superseded by:** N/A
+
+**Supersedes:** N/A
+
+**Related:** rfc-0003: developer utils, which also support viewing stream data from the earliest offset.
+
+
+## Problem
+Sometimes users want to replay data through functions, maybe because the function logic has changed, or maybe because downstream systems have lost state and that state needs to be re-created. The current Processor concept in riff doesn't support such a mechanism though, and always consumes data from the (current) end of streams.
+
+### Anti-Goals
+
+## Solution
+With a top to bottom view of the user experience, the proposed solution is to
+- introduce an offset option on each input binding when a processor is created
+- persist that option in the Processor CRD instance on the cluster
+- have the Processor reconciler forward that option to the streaming-processor, so that it changes its SubscriptionRequests accordingly
+
+### User Impact
+This RFC proposes to change the `Inputs` field of the `Processor` CRD from
+```go
+Inputs []StreamBinding `json:"inputs"`
+```
+to
+```go
+Inputs []InputStreamBinding `json:"inputs"`
+...
+type InputStreamBinding struct {
+    Stream string `json:"stream"`
+    Alias string `json:"alias,omitempty"`
+
+	// Where to start consuming this stream the first time a processor runs.
+    StartOffset string `json:"startOffset"`
```

Comment:
Something like [IntOrString](https://godoc.org/k8s.io/apimachinery/pkg/util/intstr#IntOrString) may be appropriate here, unless we really need the capacity of int64 vs int32.